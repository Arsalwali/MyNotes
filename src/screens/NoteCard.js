import React from 'react';
import { View, Text, StyleSheet, ScrollView } from 'react-native';
import { Icon } from 'react-native-elements';
import { colors, metrics, iconTypes, iconNames } from '../theme';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    borderRadius: 10,
    margin: 10,
    elevation: 4,
  },
  headerContainer: {
    flex: 1,
    flexDirection: 'row',
    borderTopLeftRadius: 10,
    borderTopRightRadius: 10,
    backgroundColor: colors.white,
    justifyContent: 'center',
    alignItems: 'center',
    height: 50,
    elevation: 2,
    padding: 10,
  },
  iconContainer: {
    width: 60,
    flexDirection: 'row',
    justifyContent: 'space-between',
  },
  innerContainer: {
    flex: 1,
    padding: 10,
    minHeight: 50,
    justifyContent: 'center',
  },
  titleStyle: {
    fontSize: metrics.title,
    fontWeight: 'bold',
    alignSelf: 'center',
    color: colors.primary1,
  },
  notesStyle: {
    fontSize: metrics.bodyText,
    color: colors.darkText,
  }
});

function NoteCard(props) {
  const { title, note, color } = props.data;
  const addColor = () => {
    const isColorExist = colors.colorPickerArray.find(col => col === color.toLowerCase());
    const noteColor = isColorExist ? color : colors.noteDefaultColor;
    return { 
      backgroundColor: noteColor,
    };
  };

  return(
    <View style={[styles.container, addColor()]}>
      <View style={styles.headerContainer}>
        <View style={{ flex: 1 }}>
          <Text style={styles.titleStyle} numberOfLines={1}>{title}</Text>
        </View>
        <View style={styles.iconContainer}>
          <Icon
            name={iconNames.edit}
            type={iconTypes.MaterialIcons}
            size={metrics.iconSize}
            color={colors.iconColor}
            onPress={props.onEditPress}
          />
          <Icon
            name={iconNames.delete}
            type={iconTypes.MaterialIcons}
            size={metrics.iconSize}
            color={colors.iconColor}
            onPress={props.onDeletePress}
          />
        </View>
      </View>
      <View style={styles.innerContainer}>
        <Text style={styles.notesStyle}>{note}</Text>
      </View>
    </View>
  );
}

export default NoteCard;