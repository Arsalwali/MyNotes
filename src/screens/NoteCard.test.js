import React from 'react';
import { shallow } from 'enzyme';
import NoteCard from './NoteCard';


describe('NoteCard component', () => {
  it('should render properly', () => {
    const props = {
      data: {
        title: 'Note Title',
        note: 'This is my first note',
        color: '#5d7a90',
      },
      onEditPress: jest.fn(),
      onDeletePress: jest.fn(),
    };
    
    const wrapper = shallow(<NoteCard {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render properly with defualt color if color not found', () => {
    const props = {
      data: {
        title: 'Note Title',
        note: 'This is my first note',
        color: '#ffffff',
      },
      onEditPress: jest.fn(),
      onDeletePress: jest.fn(),
    };

    const wrapper = shallow(<NoteCard {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
