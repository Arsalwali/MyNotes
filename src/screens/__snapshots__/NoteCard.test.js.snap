// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoteCard component should render properly 1`] = `
<View
  style={
    Array [
      Object {
        "borderRadius": 10,
        "elevation": 4,
        "flex": 1,
        "justifyContent": "center",
        "margin": 10,
      },
      Object {
        "backgroundColor": "#5d7a90",
      },
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#ffffff",
        "borderTopLeftRadius": 10,
        "borderTopRightRadius": 10,
        "elevation": 2,
        "flex": 1,
        "flexDirection": "row",
        "height": 50,
        "justifyContent": "center",
        "padding": 10,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <Text
        numberOfLines={1}
        style={
          Object {
            "alignSelf": "center",
            "color": "#517fa4",
            "fontSize": 25,
            "fontWeight": "bold",
          }
        }
      >
        Note Title
      </Text>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "width": 60,
        }
      }
    >
      <Icon
        color="#517fa4"
        name="edit"
        onPress={[MockFunction]}
        raised={false}
        reverse={false}
        reverseColor="white"
        size={28}
        type="materialicon"
        underlayColor="white"
      />
      <Icon
        color="#517fa4"
        name="delete"
        onPress={[MockFunction]}
        raised={false}
        reverse={false}
        reverseColor="white"
        size={28}
        type="materialicon"
        underlayColor="white"
      />
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "justifyContent": "center",
        "minHeight": 50,
        "padding": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#212121",
          "fontSize": 17.5,
        }
      }
    >
      This is my first note
    </Text>
  </View>
</View>
`;

exports[`NoteCard component should render properly with defualt color if color not found 1`] = `
<View
  style={
    Array [
      Object {
        "borderRadius": 10,
        "elevation": 4,
        "flex": 1,
        "justifyContent": "center",
        "margin": 10,
      },
      Object {
        "backgroundColor": "#f5fcff",
      },
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#ffffff",
        "borderTopLeftRadius": 10,
        "borderTopRightRadius": 10,
        "elevation": 2,
        "flex": 1,
        "flexDirection": "row",
        "height": 50,
        "justifyContent": "center",
        "padding": 10,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <Text
        numberOfLines={1}
        style={
          Object {
            "alignSelf": "center",
            "color": "#517fa4",
            "fontSize": 25,
            "fontWeight": "bold",
          }
        }
      >
        Note Title
      </Text>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "width": 60,
        }
      }
    >
      <Icon
        color="#517fa4"
        name="edit"
        onPress={[MockFunction]}
        raised={false}
        reverse={false}
        reverseColor="white"
        size={28}
        type="materialicon"
        underlayColor="white"
      />
      <Icon
        color="#517fa4"
        name="delete"
        onPress={[MockFunction]}
        raised={false}
        reverse={false}
        reverseColor="white"
        size={28}
        type="materialicon"
        underlayColor="white"
      />
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "justifyContent": "center",
        "minHeight": 50,
        "padding": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#212121",
          "fontSize": 17.5,
        }
      }
    >
      This is my first note
    </Text>
  </View>
</View>
`;
